### PR TITLE
OOP School Library - Preserve Data

### PR DESCRIPTION
**Pull request by DammyShittu**
## In this milestone, we completed the following requirements;

- On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. rentals.json)

- On startup of your app load all data from files. Remember to handle the case when one or more of the files are missing.

### Linters were checked and all offences were fixed.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -11,4 +11,16 @@ def initialize(title:, author:)
...
+    {
+      JSON.create_id => self.class.name,
+      'title' => @title,
+      'author' => @author
+    }.to_json(*args)
+  end
+
+  # def self.json_create(object)
+  #   new(object['title'], object['author'])
+  # end
```
Comment (Dipeshtwis) [inline book.rb]: - [ ] I would recommend removing the commented code to make the codebase cleaner

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @DammyShittu and @ridaarif98, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: Your data is preserved in JSON file and able to access in the next run as well. well done.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

- [ ] I would recommend writing the step on how to run the project. [`ruby main.rb`]

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (DammyShittu) [reply book.rb]: Hi @Dipeshtwis 
Thank you for the review.
We will work on it as soon as possible.
Cheers 👍

**Comment by helman101** [comment]
# STATUS: APPROVED 🟢 🥳 

Hi @DammyShittu 🥇 , 

### Congratulations!!! 🥳 🎉

You have successfully completed all the expectations for this project, there is nothing more to do here, now you can go on your way and keep improving your skills.

Cheers and Happy coding!👏👏👏

gif

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by DammyShittu** [comment]
Thank you @helman101 🎉

**Pull request merged**